doc/BorderRouter.md

```diff
@@ -0,0 +1,118 @@
+packet will not be processed at all. In this case the buffer is immediately returned to the receiver.
+Once a packet is processed, it gets enqueued to the forwarder which is responsible for the egress interface.
+If the queue of the forwarder is full, the packet will be dropped.
+* **Forwarders** There exists one forwarder per border router interface that is responsible for forwarding the
+packets over the network that it receives from the processing routines. It forwards the packets as a batch.
+Afterwards it returns each buffer to the receiver from which it originates.
+
+![Border Router Design](fig/border_router/br_design.png)
+
+## Mapping of processing routines
+To prevent any packet reordering on the fast-path, we map the tuple of source and flowID, see the
+[SCION header documentation](https://github.com/scionproto/scion/blob/master/doc/protocols/scion-header.rst),
+to a fixed processing routine using a hash function together with a salt value to prevent pre-computations
+of the exact mapping.
+
+## Slow path
+During processing, packets that have to follow the slow path are identified and forwarded to the
+slow-path processing routines.
+Rate limiting of slow-path operations is not implemented explicitly, but only implictily through specifying
+the number of slow-path processing routines in the configuration.
+In case a packet is identified to belong to the slow path but the queue of the slow path is full, the
+packet is dropped.
+Packets currently identified for slow-path are:
+* Malformed packets
+* SCMP traceroute packets
+
+# Configuration
+The configuration of the border router will remain in the border router toml file.
+The following configuration entries are added:
+
+## Pool size of packet buffers
+Since a pool of packet buffers is bound to a receiver, the size of the pool can be configured for each
+receiver seperately.
+It makes sense to make this configurable on a per border router interface level because not every 
+interface might have the same expected load and hence not every interface requires the same pool size.
+An appropriate default value would have to be evaluated.
+
+## Number of processing routines (N)
+By configuring the number of processing routines one can specify the number of goroutines that are able
+to process packets in parallel.
+A default value could be 4 to 8 times the number of border router interfaces.
+
+## Number of slow-path processing routines (M)
+By configuring the number of slow-path processing routines one can specify the number of goroutines that
+process the packets on the slow-path.
+A default value could be the same number as we have border router interfaces.
+
+## Processing packets channel size
+Since each processing routine has a queue of packets to process and all packets not fitting in the queue
+are dropped, one has to specify a reasonable queue size.
+A default value could be 256.
+
+# Considerations for future work
+## Multiple Receivers per Border Router interface
```

Comment:
Multiple receivers per border router interface

Be consistent about upper/lower case in the titles